script/testing/util/test_server.py

```diff
@@ -153,6 +159,20 @@ def run_test(self, test_case: TestCase):
         # run the pre test tasks
         test_case.run_pre_test()
 
+        # start a thread to collect the memory info if needed
+        if self.collect_mem_info:
+            # spawn a thread to collect memory info
+            self.collect_mem_thread = threading.Timer(
```

Comment:
This feels like it might help to break into a class or function. I was thinking class just cause I was thinking you instantiate it with the collection frequency int and it creates this timer and then you call `start` and it will gather all the data and store it in something like a `results` attribute and then you can have functions like `get_average_rss()` `get_average_vms()` `populate_incremental_metrics(incremental_oltpbench_results_arr)`. I don't think it needs to be a class but its just how I was thinking about it.

Reply:
Created a `MemoryMetrics` class for this purpose